### ISSUE TITLE
Support "Download plot" button in matplotlib widget plots in console sessions

### ISSUE THREAD

**Issue by seeM** [posit-dev/positron#4236]
## Steps to reproduce the issue:

1. In the console, create a matplotlib widget plot:

    ```py
    %matplotlib ipympl
    import matplotlib.pyplot as plt
    import numpy as np
    
    fig, ax = plt.subplots()
    
    
    x = np.linspace(0, 2*np.pi, 100)
    y = np.sin(3*x)
    ax.plot(x, y)
    ```
2. Click the "Download plot button". Nothing will happen.
<img width="126" alt="image" src="https://github.com/user-attachments/assets/b6db119a-6316-429f-bb6e-2118f1143cc0">

## Expected or desired behavior:
Positron should display a file dialog and save the file.

## Were there any error messages in the UI, Output panel, or Developer Tools console?
Nope.

---

`ipympl` handles this by creating an `<a>` element with a link to a data URI containing the base64 encoded image png, then calling its `.click` method. The webview preload script forwards this as a `clicked-data-url` event, which we're not yet handling in the `NotebookOutputWebview`. We could hook that up to the plots service's logic for saving plots.

It already works in notebooks since the `BackLayerWebview` already handles `clicked-data-url` events.

**Comment by nstrayer** [comment]
This got inadvertently fixed in #4377

**Comment by jonvanausdeln** [comment]
### Verified Fixed
Positron Version(s) : [2024.08.0-63](https://github.com/posit-dev/positron-builds/releases/tag/2024.08.0-63)
OS Version(s) : Windows 11

### Test scenario(s) 



File downloads as expected now

### Link(s) to TestRail test cases run or created:
